pkg/hintrunner/zero/zerohint_dictionaries.go

```diff
@@ -125,6 +127,62 @@ func createDictReadHinter(resolver hintReferenceResolver) (hinter.Hinter, error)
 	return newDictReadHint(dictPtr, key, value), nil
 }
 
+// DictSquashCopyDict hint prepares arguments for creating a new dictionary
+// with `dict_new` by copying the existing one in the context of squashing
+//
+// `newDictSquashCopyDictHint` takes 1 operander as argument
+//   - `dict_accesses_end` variable is a memory address that allows access to the dictionary
+//
+// `newDictSquashCopyDictHint`assigns `initial_dict` to the retrieved dictionary in the current scope
+func newDictSquashCopyDictHint(dictAccessesEnd hinter.ResOperander) hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "DictSquashCopyDict",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> # Prepare arguments for dict_new. In particular, the same dictionary values should be copied
+			//> # to the new (squashed) dictionary.
+			//> vm_enter_scope({
+			//> 	# Make __dict_manager accessible.
+			//> 	'__dict_manager': __dict_manager,
+			//> 	# Create a copy of the dict, in case it changes in the future.
+			//> 	'initial_dict': dict(__dict_manager.get_dict(ids.dict_accesses_end)),
+			//> })
+
+			dictionaryManager_, err := ctx.ScopeManager.GetVariableValue("__dict_manager")
+			if err != nil {
+				return err
+			}
+
+			dictionaryManager, ok := dictionaryManager_.(hinter.ZeroDictionaryManager)
+			if !ok {
+				return fmt.Errorf("cannot cast dictionaryManager to a DicttionaryManager struct")
+			}
+
+			dictAccessesEnd_, err := hinter.ResolveAsAddress(vm, dictAccessesEnd)
+			if err != nil {
+				return err
+			}
+
+			dictionaryCopy, err := dictionaryManager.GetDictionary(*dictAccessesEnd_)
+			if err != nil {
+				return err
+			}
+
+			ctx.ScopeManager.EnterScope(map[string]any{"__dict_manager": dictionaryManager, "initial_dict": dictionaryCopy})
```

Comment:
Just to confirm:
1. the same dictionary manager in the current scope must be available in the child scope. 
2. a copy of the `dict_accesses_end` from the same dictionary manager must be added to the child scope.

In this case is `dictionaryCopy` creating a new copied object? 🤔

Reply:
We should add some tests for this too right?

Reply:
I agree, that at least one test, which checks if the provided values are in scope is neccessary.

Reply:
yeah i'll add tests i was waiting to do it

Reply:
@har777  you're right, i need to make a deep copy of `dictionaryCopy` (to be renamed `dictionary` ) 
for that, i need to modify a lot of code because i need to do it manually, which means accessing the fields of a `MemoryValue` which is another package and currently only lowercases and therefore not accessible